homeassistant/components/tado/climate.py

```diff
@@ -222,16 +229,21 @@ def create_climate_entity(
     if heat_temperatures is None and "temperatures" in capabilities:
         heat_temperatures = capabilities["temperatures"]
 
-    if cool_temperatures is None and heat_temperatures is None:
+    if cool_temperatures is None and heat_temperatures is None and not tado.is_x:
         _LOGGER.debug("Not adding zone %s since it has no temperatures", name)
         return None
 
-    heat_min_temp = None
-    heat_max_temp = None
-    heat_step = None
-    cool_min_temp = None
-    cool_max_temp = None
-    cool_step = None
+    heat_min_temp: float | None = None
+    heat_max_temp: float | None = None
+    heat_step: float | None = None
+    cool_min_temp: float | None = None
+    cool_max_temp: float | None = None
+    cool_step: float | None = None
+
+    if tado.is_x:
```

Comment:
Instead of having a lot of these if statements, can we have separate classes for it?